// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package privacy_sandbox.server_common.gvisor;

message CallbackReadRequest {
  repeated bytes payloads = 1;
}

message CallbackReadResponse {
  uint64 payload_size = 1;
}

message CallbackWriteRequest {
  uint64 element_size = 1;
  uint64 element_count = 2;
}

message CallbackWriteResponse {
  repeated bytes payloads = 1;
}
